src/main/java/com/thirdcc/webapp/service/impl/EventChecklistServiceImpl.java

```diff
@@ -123,6 +124,40 @@ public EventChecklistDTO update(EventChecklistDTO eventChecklistDTO) {
     @Override
     public void delete(Long id) {
         log.debug("Request to delete Checklist : {}", id);
+        EventChecklist eventChecklist = checklistRepository.findById(id)
+            .orElseThrow(() -> new BadRequestException("Event Checklist not found: "+id));
+        Event event = eventService.findEventByIdAndNotCancelledStatus(eventChecklist.getEventId());
+        if (event.getStartDate().isBefore(Instant.now())) {
+            throw new BadRequestException("Cannot delete a event checklist for event that had been start");
+        }
+        if (eventChecklist.getStatus() != EventChecklistStatus.OPEN) {
+            throw new BadRequestException("Cannot delete a event Checklist that is not open status");
+        }
```

Comment:
this logic is not included in #29

if this logic is really needed, please include the it in the issue

Reply:
it should be a logic
i added in the issue